src/main/java/az/edu/turing/service/impl/BookingServiceImpl.java

```diff
@@ -34,21 +37,22 @@ private BookingDto toDto(BookingEntity booking) {
     }
 
     @Override
-    public BookingDto createBooking(long flightId, String firstName, String lastName) {
-        Optional<FlightEntity> flight = flightDao.getById(flightId);
+    public BookingDto createBooking(CreateBookingRequest createBookingRequest) {
+        Optional<FlightEntity> flight = flightDao.getById(createBookingRequest.getFlightId());
         if (!flight.isPresent()) {
-            throw new IllegalArgumentException("Flight not found with ID: " + flightId);
+            throw new FlightNotFoundException("Flight not found with ID: " + createBookingRequest.getFlightId());
         }
 
-        BookingEntity bookingEntity = new BookingEntity(0, flight.get(), firstName, lastName);
+        BookingEntity bookingEntity = new BookingEntity(flight.get(), createBookingRequest.getFirstName(),createBookingRequest.getLastName());
```

Comment:
instead of doing `if (!flight.isPresent())`, we could use .orElseThrows(), which also give us `FlightEntity` itself without optional (you wont need to use .get() )